Rubberduck.VBEEditor/WindowsApi/WM.cs

```diff
@@ -977,4 +978,242 @@ public enum WM : uint
+            {0x0216, "MOVING"},
+            {0x0218, "POWERBROADCAST"},
+            {0x0219, "DEVICECHANGE"},
+            {0x0220, "MDICREATE"},
+            {0x0221, "MDIDESTROY"},
+            {0x0222, "MDIACTIVATE"},
+            {0x0223, "MDIRESTORE"},
+            {0x0224, "MDINEXT"},
+            {0x0225, "MDIMAXIMIZE"},
+            {0x0226, "MDITILE"},
+            {0x0227, "MDICASCADE"},
+            {0x0228, "MDIICONARRANGE"},
+            {0x0229, "MDIGETACTIVE"},
+            {0x0230, "MDISETMENU"},
+            {0x0231, "ENTERSIZEMOVE"},
+            {0x0232, "EXITSIZEMOVE"},
+            {0x0233, "DROPFILES"},
+            {0x0234, "MDIREFRESHMENU"},
+            {0x0281, "IME_SETCONTEXT"},
+            {0x0282, "IME_NOTIFY"},
+            {0x0283, "IME_CONTROL"},
+            {0x0284, "IME_COMPOSITIONFULL"},
+            {0x0285, "IME_SELECT"},
+            {0x0286, "IME_CHAR"},
+            {0x0288, "IME_REQUEST"},
+            {0x0290, "IME_KEYDOWN"},
+            {0x0291, "IME_KEYUP"},
+            {0x02A1, "MOUSEHOVER"},
+            {0x02A3, "MOUSELEAVE"},
+            {0x02A0, "NCMOUSEHOVER"},
+            {0x02A2, "NCMOUSELEAVE"},
+            {0x02B1, "WTSSESSION_CHANGE"},
+            {0x02c0, "TABLET_FIRST"},
+            {0x02df, "TABLET_LAST"},
+            {0x0300, "CUT"},
+            {0x0301, "COPY"},
+            {0x0302, "PASTE"},
+            {0x0303, "CLEAR"},
+            {0x0304, "UNDO"},
+            {0x0305, "RENDERFORMAT"},
+            {0x0306, "RENDERALLFORMATS"},
+            {0x0307, "DESTROYCLIPBOARD"},
+            {0x0308, "DRAWCLIPBOARD"},
+            {0x0309, "PAINTCLIPBOARD"},
+            {0x030A, "VSCROLLCLIPBOARD"},
+            {0x030B, "SIZECLIPBOARD"},
+            {0x030C, "ASKCBFORMATNAME"},
+            {0x030D, "CHANGECBCHAIN"},
+            {0x030E, "HSCROLLCLIPBOARD"},
+            {0x030F, "QUERYNEWPALETTE"},
+            {0x0310, "PALETTEISCHANGING"},
+            {0x0311, "PALETTECHANGED"},
+            {0x0312, "HOTKEY"},
+            {0x0317, "PRINT"},
+            {0x0318, "PRINTCLIENT"},
+            {0x0319, "APPCOMMAND"},
+            {0x031A, "THEMECHANGED"},
+            {0x031D, "CLIPBOARDUPDATE"},
+            {0x031E, "DWMCOMPOSITIONCHANGED"},
+            {0x031F, "DWMNCRENDERINGCHANGED"},
+            {0x0320, "DWMCOLORIZATIONCOLORCHANGED"},
+            {0x0321, "DWMWINDOWMAXIMIZEDCHANGE"},
+            {0x033F, "GETTITLEBARINFOEX"},
+            {0x0358, "HANDHELDFIRST"},
+            {0x035F, "HANDHELDLAST"},
+            {0x0360, "AFXFIRST"},
+            {0x037F, "AFXLAST"},
+            {0x0380, "PENWINFIRST"},
+            {0x038F, "PENWINLAST"},
+            {0x8000, "APP"},
+            {0x0400, "USER"},
+            {0x1400, "CPL_LAUNCH"},
+            {0x1401, "CPL_LAUNCHED"},
+            {0x118, "SYSTIMER"},
+            {0x1300, "RUBBERDUCK_CHILD_FOCUS"},
+            {0x111E, "RUBBERDUCK_SINKING"}
+        };
+    }
+
+#endif
```

Comment:
These should probably have the prefixes on them - otherwise they're less searchable. Also, should the flags be in there (i.e. USER and APP)?

It might also be easier and more maintainable to just build the lookup from the WM enum in `Rubberduck.VBEditor.WindowsApi` - that way we don't have to maintain 2 copies and keep them in sync.

Reply:
Sure, will take a look. I just Excel-coded what was defined already, which was a bit lazy.